--- libxml2-2.14.5/include/libxml/xmlerror.h~	2025-07-15 15:08:05.000000000 +0200
+++ libxml2-2.14.5/include/libxml/xmlerror.h	2025-08-20 13:04:41.913857953 +0200
@@ -20,7 +20,7 @@ extern "C" {
  * Backward compatibility
  */
 #define initGenericErrorDefaultFunc(h) \
-    xmlSetGenericErrorFunc(NULL, (h) ? *(h) : NULL)
+    xmlSetGenericErrorFunc(NULL, (h) ? *((xmlGenericErrorFunc *) (h)) : NULL)
 
 /**
  * xmlErrorLevel:
