diff -dur libxml2-2.6.17.orig/tree.c libxml2-2.6.17/tree.c
--- libxml2-2.6.17.orig/tree.c	2005-01-04 15:49:48.000000000 +0100
+++ libxml2-2.6.17/tree.c	2005-02-28 11:53:32.101621669 +0100
@@ -5663,7 +5663,7 @@
                 if ((cur->href != NULL) && (href != NULL) &&
                     (xmlStrEqual(cur->href, href))) {
 		    if (((!is_attr) || (cur->prefix != NULL)) &&
-		        (xmlNsInScope(doc, orig, node, cur->href) == 1))
+		        (xmlNsInScope(doc, orig, node, cur->prefix) == 1))
 			return (cur);
                 }
                 cur = cur->next;
@@ -5674,7 +5674,7 @@
                     if ((cur->href != NULL) && (href != NULL) &&
                         (xmlStrEqual(cur->href, href))) {
 			if (((!is_attr) || (cur->prefix != NULL)) &&
-		            (xmlNsInScope(doc, orig, node, cur->href) == 1))
+		            (xmlNsInScope(doc, orig, node, cur->prefix) == 1))
 			    return (cur);
                     }
                 }
