--- libxml2-2.4.15/python/Makefile.am~	Tue Feb 12 17:25:11 2002
+++ libxml2-2.4.15/python/Makefile.am	Tue Feb 12 18:33:55 2002
@@ -38,7 +38,7 @@
 noinst_LTLIBRARIES = libxmlmodule.la
 libxmlmodule_la_SOURCES = libxml.c types.c libxml2-py.c
 
-libxml2mod.so: $(libxmlmodule_la_OBJECTS)
+libxml2mod.so$(EXEEXT): $(libxmlmodule_la_OBJECTS)
 	$(LINK) -o $@ $(libxmlmodule_la_OBJECTS) $(libxml2mod_so_LDFLAGS)
 
 
