--- libxml2-2.9.10/xml2-config.in.orig	2020-01-19 17:14:27.721696222 +0100
+++ libxml2-2.9.10/xml2-config.in	2020-01-19 17:22:13.425839953 +0100
@@ -87,7 +87,7 @@
             libs="@XML_LIBS@ @XML_PRIVATE_LIBS@ @MODULE_PLATFORM_LIBS@"
         fi
 
-        if [ "@XML_LIBDIR@" != "-L/usr/lib" -a "@XML_LIBDIR@" != "-L/usr/lib64" ]; then
+        if [ "@XML_LIBDIR@" != "-L/usr/lib" -a "@XML_LIBDIR@" != "-L/usr/lib64" -a "@XML_LIBDIR@" != "-L/usr/libx32" ]; then
             libs="@XML_LIBDIR@ $libs"
         fi
 
