--- libxml2-2.10.0/python/setup.py.in.orig	2022-05-02 14:10:21.000000000 +0200
+++ libxml2-2.10.0/python/setup.py.in	2022-08-18 21:39:44.799159134 +0200
@@ -73,19 +73,10 @@
 HOME
 ];
 
-xml_includes=""
-for dir in includes_dir:
-    if not missing(dir + "/libxml2/libxml/tree.h"):
-        xml_includes=dir + "/libxml2"
-        break;
-
-if xml_includes == "":
-    print("failed to find headers for libxml2: update includes_dir")
-    sys.exit(1)
-
 # those are added in the linker search path for libraries
 libdirs = [
-os.path.join(ROOT,'lib'),
+'@abs_top_builddir@/.libs',
+'@libdir@',
 ]
 
 xml_files = ["libxml2-api.xml", "libxml2-python-api.xml",
@@ -194,7 +185,7 @@
 if WITHDLLS:
     modules.append('libxmlmods.__init__')
 c_files = ['libxml2-py.c', 'libxml.c', 'types.c' ]
-includes= [xml_includes]
+includes= ['@abs_top_builddir@/include']
 libs    = [libraryPrefix + "xml2"] + platformLibs
 macros  = []
 if with_threads:
