diff -urN libxml2-2.5.10.orig/Makefile.am libxml2-2.5.10/Makefile.am
--- libxml2-2.5.10.orig/Makefile.am	Fri Aug 15 15:03:34 2003
+++ libxml2-2.5.10/Makefile.am	Fri Aug 15 15:06:34 2003
@@ -5,7 +5,7 @@
 
 DIST_SUBDIRS = include . doc example python
 
-INCLUDES = -I$(top_builddir)/include -I@srcdir@/include @THREAD_CFLAGS@ @Z_CFLAGS@ 
+INCLUDES = -I$(top_builddir)/include -I$(srcdir)/include $(THREAD_CFLAGS) $(Z_CFLAGS)
 
 noinst_PROGRAMS=testSchemas testRelax testSAX testHTML testXPath testURI \
                 testDocbook testThreads testC14N testAutomata testRegexp \
@@ -16,9 +16,9 @@
 bin_SCRIPTS=xml2-config
 
 lib_LTLIBRARIES = libxml2.la
-libxml2_la_LIBADD = @THREAD_LIBS@ @Z_LIBS@ $(ICONV_LIBS) @M_LIBS@ @WIN32_EXTRA_LIBADD@
+libxml2_la_LIBADD = $(THREAD_LIBS) $(Z_LIBS) $(ICONV_LIBS) $(M_LIBS) $(WIN32_EXTRA_LIBADD)
 
-libxml2_la_LDFLAGS = @WIN32_EXTRA_LDFLAGS@ -version-info @LIBXML_VERSION_INFO@
+libxml2_la_LDFLAGS = $(WIN32_EXTRA_LDFLAGS) -version-info $(LIBXML_VERSION_INFO)
 
 if WITH_TRIO_SOURCES
 libxml2_la_SOURCES = SAX.c entities.c encoding.c error.c parserInternals.c  \
@@ -39,7 +39,7 @@
 endif
 
 DEPS = $(top_builddir)/libxml2.la
-LDADDS = @STATIC_BINARIES@ $(top_builddir)/libxml2.la @THREAD_LIBS@ @Z_LIBS@ $(ICONV_LIBS) @M_LIBS@ @WIN32_EXTRA_LIBADD@
+LDADDS = $(STATIC_BINARIES) $(top_builddir)/libxml2.la $(THREAD_LIBS) $(Z_LIBS) $(ICONV_LIBS) $(M_LIBS) $(WIN32_EXTRA_LIBADD)
 
 
 man_MANS = xml2-config.1 libxml.3
@@ -50,7 +50,7 @@
 xmllint_SOURCES=xmllint.c
 xmllint_LDFLAGS = 
 xmllint_DEPENDENCIES = $(DEPS)
-xmllint_LDADD=  @RDL_LIBS@ $(LDADDS)
+xmllint_LDADD=  $(RDL_LIBS) libxml2.la
 
 testSAX_SOURCES=testSAX.c
 testSAX_LDFLAGS = 
@@ -789,8 +789,6 @@
 
 CLEANFILES=xml2Conf.sh
 
-confexecdir=$(libdir)
-confexec_DATA = xml2Conf.sh
 EXTRA_DIST = xml2-config.in xml2Conf.sh.in libxml.spec.in libxml2.spec \
              libxml.m4 Copyright check-xml-test-suite.py \
 	     check-relaxng-test-suite.py check-xinclude-test-suite.py \
diff -urN libxml2-2.5.10.orig/include/libxml/Makefile.am libxml2-2.5.10/include/libxml/Makefile.am
--- libxml2-2.5.10.orig/include/libxml/Makefile.am	Fri Aug 15 15:03:34 2003
+++ libxml2-2.5.10/include/libxml/Makefile.am	Fri Aug 15 15:04:18 2003
@@ -41,7 +41,4 @@
 		xmlreader.h \
 		relaxng.h
 
-install-exec-hook:
-	$(mkinstalldirs) $(DESTDIR)$(xmlincdir)
-
 EXTRA_DIST = xmlversion.h.in
