diff -Nur libxml2-2.4.16.orig/python/Makefile.am libxml2-2.4.16/python/Makefile.am
--- libxml2-2.4.16.orig/python/Makefile.am	Mon Feb 18 04:35:24 2002
+++ libxml2-2.4.16/python/Makefile.am	Thu Mar  7 14:11:55 2002
@@ -38,7 +38,7 @@
 noinst_LTLIBRARIES = libxmlmodule.la
 libxmlmodule_la_SOURCES = libxml.c types.c libxml2-py.c
 
-libxml2mod.so: $(libxmlmodule_la_OBJECTS) $(mylibs)
+libxml2mod.so$(EXEEXT): $(libxmlmodule_la_OBJECTS) $(mylibs)
 	$(LINK) -o $@ $(libxmlmodule_la_OBJECTS) $(libxml2mod_so_LDFLAGS)
 
 
